test/rng_testsuite/statistics_tests/exponential_distribution_test.pass.cpp

```diff
@@ -0,0 +1,638 @@
+#include <iostream>
+
+#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
+#    include <CL/sycl.hpp>
+#    include <random>
+#    include <limits>
+#    include <oneapi/dpl/random>
+#    include <math.h>
+
+// Engine parameters
+constexpr auto a = 40014u;
+constexpr auto c = 200u;
+constexpr auto m = 2147483563u;
+constexpr auto seed = 777;
+
+template <typename ScalarRealType>
+int
+statistics_check(int nsamples, ScalarRealType lambda, const std::vector<ScalarRealType>& dpstd_samples)
+{
+    // theoretical moments
+    double tM = 1 / lambda;
+    double tD = 1 / (lambda * lambda);
+    double tQ = 9 / (lambda * lambda * lambda * lambda);
+
+    // sample moments
+    double sum = 0.0;
+    double sum2 = 0.0;
+    for (int i = 0; i < nsamples; i++)
+    {
+        sum += dpstd_samples[i];
+        sum2 += dpstd_samples[i] * dpstd_samples[i];
+    }
+    double sM = sum / nsamples;
+    double sD = sum2 / nsamples - sM * sM;
+
+    // comparison of theoretical and sample moments
+    double tD2 = tD * tD;
+    double s = ((tQ - tD2) / nsamples) - (2 * (tQ - 2.0 * tD2) / (nsamples * nsamples)) +
+               ((tQ - 3.0 * tD2) / (nsamples * nsamples * nsamples));
+
+    double DeltaM = (tM - sM) / sqrt(tD / nsamples);
+    double DeltaD = (tD - sD) / sqrt(s);
+
+    if (fabs(DeltaM) > 3.0 || fabs(DeltaD) > 3.0)
+    {
+        std::cout << "Error: sample moments (mean= " << sM << ", variance= " << sD
+                  << ") disagree with theory (mean=" << tM << ", variance= " << tD << ")";
+        return 1;
+    }
+
+    return 0;
+}
+
+template <class RealType, class UIntType>
+int
+test(oneapi::dpl::internal::element_type_t<RealType> lambda, int nsamples)
+{
+
+    sycl::queue queue(sycl::default_selector{});
+
+    // memory allocation
+    std::vector<oneapi::dpl::internal::element_type_t<RealType>> dpstd_samples(nsamples);
+
+    constexpr int num_elems = oneapi::dpl::internal::type_traits_t<RealType>::num_elems == 0
+                                  ? 1
+                                  : oneapi::dpl::internal::type_traits_t<RealType>::num_elems;
+
+    // dpstd generation
+    {
+        sycl::buffer<oneapi::dpl::internal::element_type_t<RealType>, 1> dpstd_buffer(dpstd_samples.data(), nsamples);
+
+        queue.submit([&](sycl::handler& cgh) {
+            auto dpstd_acc = dpstd_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+            cgh.parallel_for<>(sycl::range<1>(nsamples / num_elems), [=](sycl::item<1> idx) {
+                unsigned long long offset = idx.get_linear_id() * num_elems;
+                oneapi::dpl::linear_congruential_engine<UIntType, a, c, m> engine(seed, offset);
+                oneapi::dpl::exponential_distribution<RealType> distr(lambda);
+
+                sycl::vec<oneapi::dpl::internal::element_type_t<RealType>, num_elems> res = distr(engine);
+                res.store(idx.get_linear_id(), dpstd_acc.get_pointer());
+            });
+        });
+        queue.wait();

```

Comment:
I think we don't need wait since the buffer is created in the scope. When the buffer is destroyed, synchronization will be done automatically.

Reply:
done